tools.yml

```
@@ -1053,7 +1053,19 @@ tools:
   toolshed.g2.bx.psu.edu/repos/iuc/anndata_export/anndata_export/.*:
     cores: 4
   toolshed.g2.bx.psu.edu/repos/iuc/anndata_import/anndata_import/.*:
-    mem: 200
+    rules:
+      - if: input_size < 0.05
+        mem: 1
+      - if: 0.05 <= input_size < 0.2
+        mem: 4
+      - if: 0.2 <= input_size < 0.5
+        mem: 16
+      - if: 0.5 <= input_size < 1
+        mem: 32
+      - if: 1 <= input_size < 5
+        mem: 100
+      - if: input_size >= 5
+        fail: Too much data, please check if the input is correct.
```

Does the tool really not work with inputs bigger than 5GB? Do you think it would make more sense to use something similar to what is shown below?

```suggestion
      - if: input_size >= 5
        mem: 20 * input_size
```

Remember that with the current rule, you are _forbidding_ users from running the tool on inputs bigger than 5GB. Even if you are unsure of how much memory may be needed, a tool that crashes _sometimes_ for inputs above 5GB is better than a tool that cannot be used at all for inputs above 5GB (no matter how powerful your cluster is).

If the tool really does not work or does not make sense to be used for inputs bigger than 5GB then it is ok to keep things as they are.

---

Probably,  `10 * input_size` works most of the times. What happens if the job fails because the memory is not enough? Will it be resubmitted with a larger memory?

---

> 

No, it will just fail. However, I think there are resubmit rules too. @nuwang might know more. However, I want to remind y'all:

Resubmission > 10 * input_size (with potential crashes) > Forbidding use

---

> Probably, `10 * input_size` works most of the times. What happens if the job fails because the memory is not enough? Will it be resubmitted with a larger memory?

Then I'd say commit `10 * input_size` unless @nuwang has a better idea. The resubmission thing I was talking about is [this](https://github.com/galaxyproject/total-perspective-vortex/pull/78/files).

---

That sounds good to me. There are some issues with resubmission with dynamic destinations that need to be fixed on the Galaxy end: https://github.com/galaxyproject/galaxy/issues/15208

---

I'm trying to figure out how much memory is required for `anndata_inspect` because it has an arbitrary 15.2G mem requirement. It is a bit strange that some jobs of `anndata_inspect` are still running since yesterday. I will update the memory requirement for both tools as soon as I get the results.

---

It seems `anndata_inspect` is not memory dependent. I tried on different-sized datasets. With less memory, it took more time to run but did not fail. Current 15.2G seems reasonably good for most of the inspect functions. So no I just updated the memory of `anndata_import` to `10 * input_size`.